File: src/main/java/com/example/smarthouse/sensor/security/SecuritySensor.java
package com.example.smarthouse.sensor;

import com.example.smarthouse.sensor.config.SensorProtocol;
import com.example.smarthouse.sensor.config.SensorState;
import com.example.smarthouse.sensor.config.SensorType;
import com.example.smarthouse.sensor.message.MessageType;
import com.example.smarthouse.sensor.message.SensorMessage;
import lombok.Data;

import java.util.Date;

public class SecuritySensor extends DefaultSensor {
    private boolean alarmActivated;

    public SecuritySensor(int sensorId, String sensorName, SensorProtocol sensorProtocol) {
        super(sensorId, sensorName, SensorState.ON, sensorProtocol, SensorType.SECURITY_SENSOR);
        this.alarmActivated = false;
    }

    public void activateAlarm() {
        alarmActivated = true;
        System.out.println("Alarm activated");
    }

    public void deactivateAlarm() {
        alarmActivated = false;
        System.out.println("Alarm deactivated");
    }

    @Override
    public SensorMessage getDataFromSensor() {
        return new SensorMessage(
                getSensorId(),
                alarmActivated ? 1 : 0,
                new Date(),
                MessageType.SENSOR_DATA,
                alarmActivated ? "Alarm activated" : "Alarm deactivated",
                getSensorType()
        );
    }
}

File: src/main/kotlin/com/example/sensor/SecuritySensor.kt
import com.example.config.SensorProtocol
import com.example.config.SensorState
import com.example.config.SensorType
import com.example.message.MessageType
import com.example.message.Message
import com.example.smarthouse.sensor.DefaultSensor
import java.util.*

class SecuritySensor(sensorId: Int, sensorName: String?, sensorProtocol: SensorProtocol?) :
    DefaultSensor(sensorId, sensorName, SensorState.ON, sensorProtocol, SensorType.SECURITY_SENSOR) {
    private var alarmActivated = false
    fun activateAlarm() {
        alarmActivated = true
        println("Alarm activated")
    }

    fun deactivateAlarm() {
        alarmActivated = false
        println("Alarm deactivated")
    }

    override val dataFromSensor: Message
        get() = Message(
            getSensorId(),
            if (alarmActivated) 1 else 0,
            Date(),
            MessageType.SENSOR_DATA,
            if (alarmActivated) "Alarm activated" else "Alarm deactivated",
            getSensorType()
        )
}